routing/index_graph.hpp

```
@@ -77,16 +187,169 @@ class IndexGraph final
+  void ApplyRestrictionOnly(RestrictionInfo const & restrictionInfo);
+
+  /// \brief Adds restriction to navigation graph which says that it's prohibited to go from
+  /// |restrictionPoint.m_from| to |restrictionPoint.m_to|.
+  /// \note |from| and |to| could be only begining or ending feature points and they has to belong
+  /// to the same junction with |jointId|. That means features |from| and |to| has to be adjacent.
+  /// \note This method could be called only after |m_roadIndex| have been loaded with the help of
+  /// Deserialize() or Import().
+  void ApplyRestrictionNoRealFeatures(RestrictionPoint const & restrictionPoint);
+
+  /// \brief Adds restriction to navigation graph which says that from feature
+  /// |restrictionPoint.m_from| it's permitted only
+  /// to go to feature |restrictionPoint.m_to|. All other ways starting form
+  /// |restrictionPoint.m_form| is prohibited.
+  /// \note All notes which are valid for ApplyRestrictionNoRealFeatures()
+  /// is valid for ApplyRestrictionOnlyRealFeatures().
+  void ApplyRestrictionOnlyRealFeatures(RestrictionPoint const & restrictionPoint);
+
+  void ApplyRestrictionRealFeatures(RestrictionPoint const & restrictionPoint,
+                                    function<void(RestrictionInfo const &)> && f);
+
+  /// \brief Add restrictions in |restrictions| to |m_ftPointIndex|.
+  void ApplyRestrictions(RestrictionVec const & restrictions);
+
+  /// \brief Fills |path| with points from point |from| to point |to|
+  /// \note |from| and |to| should belong to the same feature.
+  /// \note The order on points in items of |connectionPaths| is from |from| to |to|.
+  void GetSingleFeaturePath(RoadPoint const & from, RoadPoint const & to,
+                            vector<RoadPoint> & path);
+
+  /// \brief  Fills |connectionPaths| with all path from joint |from| to joint |to|.
+  /// If |from| and |to| don't belong to the same feature |connectionPaths| an exception
+  /// |RoutingException| with be raised.
```

Why do we need to throw an exception? Why not to just return false?